label_studio/ml_models/serializers.py

```diff
@@ -16,19 +17,103 @@ class CreatedByFromContext:
     def __call__(self, serializer_field):
         return serializer_field.context.get('created_by')
 
-
 class ModelInterfaceSerializer(serializers.ModelSerializer):
     created_by = UserSimpleSerializer(default=CreatedByFromContext(), help_text='User who created Dataset')
+ 
+    class Meta:
+        model = ModelInterface
+        fields = '__all__'
+        read_only_fields = ['created_by', 'created_at', 'updated_at']
+
+    def check_output_classes(self, project, provided_output_classes):
+        labels, _ = get_all_labels(project.label_config)
+        project_output_classes =  sorted(list(set([label for label_list in labels.values() for label in label_list])))
+        print(project_output_classes)
+        print(provided_output_classes)
+        if project_output_classes != provided_output_classes:
+            raise ValidationError(f'output_classes not compatible with Project (id:{project.pk})')
+        
+
+    def check_input_fields(self, project, provided_input_fields):
+        parsed_config = project.get_parsed_config()
+        project_input_fields = set()
+        for tag in parsed_config:
+            for input in parsed_config[tag]['inputs']:
+                if input.get('type',None) and input.get('value', None):
+                    project_input_fields.add(input.get('value'))
+        
+        project_input_fields = sorted(list(project_input_fields))
+        print(project_input_fields, provided_input_fields)
+        if project_input_fields != provided_input_fields:
+            raise ValidationError(f'input_fields do not match inputs in Project (id:{project.pk})')
+        
+    def validate(self, data):        
+        model_obj = getattr(self, 'instance', None)
+        if model_obj:
+            if 'input_fields' in data:
+                provided_input_fields = sorted(data['input_fields'])
```

Comment:
If we're just sorting these for the equality check, can we just put them in a `set` instead? Imo that makes what we're doing slightly more clear, plus should be a bit faster (not that I expect this to be a bottleneck)